tests/test_copy.py

```diff
@@ -68,6 +68,130 @@ def test_copy_with_non_version_tags(tmp_path_factory: pytest.TempPathFactory) ->
     )
 
 
+def test_copy_with_local_version_tags_PEP440_post(
+    tmp_path_factory: pytest.TempPathFactory,
+) -> None:
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+    build_file_tree(
+        {
+            (src / "copier.yaml"): (
+                """\
+                _tasks:
+                    - cat v1.txt
+                """
+            ),
+            (src / "v1.txt"): "file only in v1",
+        }
+    )
+    with local.cwd(src):
+        git("init")
+        git("add", ".")
+        git("commit", "-m1")
+        git("tag", "1.2.3")
+
+    with local.cwd(src):
+        Path("v1.txt").write_text("Test content2")
+        git("init")
+        git("add", ".")
+        git("commit", "-m1")
+        git("tag", "1.2.3.post1")
+
+    with local.cwd(src):
+        Path("v1.txt").write_text("Test content3")
+        git("add", ".")
+        git("commit", "-m1")
+
+    copier.run_auto(
+        str(src),
+        dst,
+        vcs_ref="HEAD",
```

Comment:
If you pass this, you skip proper tag auto-detection! 

Instead, don't pass this arg and assert it picks the correct commit when copying.

Reply:
That was the intented test behavior :-/.

If I remove vcs_ref="HEAD", copier will take the previous fully versionned tag.

sample:
commit n1 -> tag 1.1.1           
  -> copier from here ok, both with ou without HEAD
commit n2 -> tag 1.1.1.dev1 
  -> copier from here ok, without HEAD it will take the previous one, with HEAD this one.
commit n3 -> !! no tag           
 -> copier from here ok, but only if no HEAD

This pull request try to allow the last one to work too.
With this patch, the last one is recognized as tag 1.1.1.dev2+XXXXX, which is okay.
WIthout, this patch, copier raise an exception.

Only drawback is for "post" tags that would become: 1.1.1.pos1.dev1+XXXXX, would be better 1.1.1.post2+XXXXX, but it is still PEP440 compliant :)  So its another feature that can be for another later update... !

Is it ok for you if I keep this parameter, but add checking on copier stdout ?
I actually wanted to add that at previous push, but I do not know if relying on sdtdout was good ... as it doesnt aim to be a standardized interface I guess ?

Reply:
AFAICS the main point of this test is the fact that the tag doesn't have a `v` prefix. Sorry, it wasn't very obvious. I ran the tests without the solution and indeed they fail. Thanks!